### PR TITLE
Encapsulate Redux selector in PublicConfigHeader and PublicConfigFooter

### PR DESCRIPTION
**Pull request by pedrottimark**
### Objective

As a prerequisite to fix layout problems with banners:
* Replace various data patterns with consistent encapsulation within components.
* Therefore, simplify MainPage and minimize risk of re-rendering pages.

### Changed files

Thanks to Dave for encouraging and helping me to rewrite the reducers file in TypeScript.

1. Edit cypress/integration/systemConfig/systemConfig.selectors.js
    * Replace `data-testid` selector as in changes 7 and 8 below. Although public config header and footer are implemented as PatternFly `Banner` elements, distinguish from other banners in folder structure and naming convention, because of their different place in DOM layout.

2. Delete src/Components/AppBanner.tsx
    * Superseded by components in changes 7 and 8 below.

3. Edit src/Containers/AppWrapper.tsx
    * Remove `publicConfig` prop.
    * Because only `MainPage` provided the prop for conditional rendering, move responsibility to render the element.

4. Edit src/Containers/Login/LoginNotice.js
    * Replace `publicConfig` with more specific `loginNotice` prop.

5. Edit src/Containers/MainPage/MainPage.tsx
    * Remove `publicConfig` selector and prop.
    * Render `PublicConfigHeader` and `PublicConfigFooter` elements.

6. Add src/Containers/MainPage/PublicConfig/PublicConfig.utils.ts
    * Factor out style from system configuration public properties.

7. Add ui/apps/platform/src/Containers/MainPage/PublicConfig/PublicConfigFooter.tsx
    * Supersede `AppBanner` and encapsulate data.

8. Add ui/apps/platform/src/Containers/MainPage/PublicConfig/PublicConfigHeader.tsx
    * Supersede `AppBanner` and encapsulate data.

9. Edit src/Containers/SystemConfig/Form/SystemConfigForm.tsx
    * Rename `systemConfig` as `publicConfig` reducer.

10. Edit src/index.tsx
    * Rename `systemConfig` as `publicConfig` reducer.

11. Edit src/reducers/index.js
    * Rename `systemConfig` as `publicConfig` reducer.

12. Edit and rename src/reducers/systemConfig.js as publicConfig.ts
    * Initialize state of reducers.
    * Replace `getPublicConfig` with more specific reducers.

### Residue

1. Replace various data patterns with consistent encapsulation within banner components.
    * `<UnreachableWarning serverState={serverState} />`
    * `{metadata?.stale && <VersionOutOfDate />}`
    * `<DatabaseBanner isApiReachable={serverState && serverState !== 'UNREACHABLE'} />`
2. Remove conditional rendering for search modal which has been superseded by search page.
3. Refactor layout to minimize `div` elements and solve problems:
    * Replace `<AppWrapper>` with fragment.
    * Move `div` in `MainPage` component to enclose PatternFly `Page` element only.
    * If possible, remove `div` in `Body` component.
4. Improve types:
    * src/utils/fetchingReduxRoutines.ts seems to export types for action creator functions, but not for corresponding action objects.
        Instead of `FetchingAction<{ response: PublicConfig }>` I wanted to write someting like `SuccessAction<PubicConfig>`
    * Specify types for thunk functions in reducers like publicConfig.
    * src/index.ts will reducer files in TypeScript remove need for the following:

        ```ts
        const dispatch = (action) =>
            (store.dispatch as ThunkDispatch<unknown, unknown, AnyAction>)(
                action as ThunkAction<void, unknown, unknown, AnyAction>
            );
        ```

## Checklist
- [x] Investigated and inspected CI test results
- [x] Updated integration tests

## Testing Performed

PublicConfigLoginNotice
LoginNotice

Verify no change in current correct and incorrect scrolling behaviors.

### Can scroll entire main page content

with `PublicConfigHeader` element only for /main/dashboard
Header-Dashboard

with `PublicConfigFooter` element only for /main/violations
Footer_Violations

with `PublicConfigHeader` and `PublicConfigFooter` elements for /main/search
Header_Footer_Search

### Cannot see some content

with `PublicConfigHeader` and `UnreachableWarning` element for /main/policy-management/policies
Header_UnreachableWarning_Policies

with `PublicConfigFooter` and `AnnouncementBanner` element for /main/integrations (can scroll the page, but footer is not visible)
Footer_Announcement_Integrations

with `PublicConfigHeader` and `PublicConfigFooter` and `CertificateExpiryBanner` elements (cannot click **Save** button)
Header-Footer-CertificateExpiryBanner

### CODE REVIEW

**Comment by pedrottimark** [comment]
Backend error unrelated to these frontend changes:

gke-postgres-ui-e2e-tests failed because **ERROR: Found at least one suspicious log file entry.**

node/datastore/store/postgres: REDACTED store.go:688: Error: Number of node component from edges (4) is unexpected (0) for node REDACTED

File: ui/apps/platform/src/reducers/publicConfig.ts [hunk 1]
```diff
@@ -30,58 +35,76 @@ export const fetchPublicConfigThunk = () => {
...
-    isLoading,
+    publicConfigError,
 });
 
 // Selectors
 
-const getPublicConfig = (state) => state.publicConfig;
-const getPublicConfigError = (state) => state.error;
-const getIsLoadingPublicConfig = (state) => state.isLoading;
+type State = ReturnType<typeof reducer>;
```
Comment (dvail) [inline ui/apps/platform/src/reducers/publicConfig.ts:94]: Even better than manually specifying the return type. 👍 

I tested this on my end for metadata I get a return type of `{ metadata: never }` due to the type I was using not actually being the exact type (an edge case I missed). Another win for defining types in terms of types. 😃

**Review by dvail — APPROVED** [review]
I'm assuming public config banners are never displayed on the login page, since they required an authenticated request?

If so, even better that those are removed from `AppWrapper` since that component is shared but will only ever display the banners in one instance.

File: ui/apps/platform/src/reducers/publicConfig.ts (hunk 1 above)
Comment (pedrottimark) [reply ui/apps/platform/src/reducers/publicConfig.ts:94]: Yes, thank you again for pointing me in a type-safer direction. There is room for improvement in follow ups.

**Pull request merged**